src/shared/utilities/collectionUtils.ts

```diff
@@ -274,6 +274,19 @@ export function applyPrimitives<T1 extends Record<string, any>, T2 extends T1>(o
     Object.assign(obj, clone)
 }
 
+/**
+ * A looser form of {@link applyPrimitives}.
```

Comment:
is there any way we can find names that are closer to each other? e.g. `assign` and `assignPrimitives` ? Helps with discoverability.

Reply:
I can get rid of `assignPrimitives`. Not really needed anymore.